scenario/mocking.py

```diff
@@ -688,19 +717,36 @@ def _layers(self) -> Dict[str, pebble.Layer]:
     def _service_status(self) -> Dict[str, pebble.ServiceStatus]:
         return self._container.service_status
 
-    def exec(self, *args, **kwargs):  # noqa: U100 type: ignore
-        cmd = tuple(args[0])
-        out = self._container.exec_mock.get(cmd)
+    # Based on a method of the same name from ops.testing.
+    def _find_exec_handler(self, command) -> Tuple[Optional[str], Optional["Exec"]]:
+        handlers = {
+            tuple(exc.command): (key, exc) for key, exc in self._container.execs.items()
```

Comment:
makes it easier to follow the code since we seem to be calling the 'key' 'exec_id' elsewhere
```suggestion
            tuple(exec.command): (exec_id, exec) for exec_id, exec in self._container.execs.items()
```